objects/input-object-array/script.js

```diff
@@ -1,22 +1,48 @@
 let alunos = [];
+const button = document.querySelector("button");
 
-function add(){
+function generateList() {
+  const lista = document.querySelector('.lista');
+
+  for (let i = 0; i < alunos.length; i++) {
+    const text = ("Nome: " + `${alunos[i].nome}` + " Nota: " + `${alunos[i].nota} `)
+
+    const item = document.createElement('li');
+    item.appendChild(document.createTextNode(text));
+    lista.appendChild(item);
+  }
+
+}
+
+
+function clearInput() { //Limpar input assim que submeter
+  document.querySelector('.nome').value = "";
+  document.querySelector('.nota').value = "";
+}
+
+
+function removeItems() {
+  document.querySelector('.lista').innerHTML = "";
+}
+
+function handleSubmit() {
   const nomeAluno = document.querySelector('.nome').value;
   const notaAluno = document.querySelector('.nota').value;
-  const resultado = document.querySelector('.resultado');
 
-  const dados = {
+  const notas = {
     nome: nomeAluno,
     nota: notaAluno
   }
-  alunos.push(dados)
 
-  clearInput()
-  resultado.innerHTML += ('O aluno ' + dados.nome + " ficou com a nota " + dados.nota)
+  alunos.push(notas)
+
 
+  clearInput()
+  removeItems()
+  generateList()
 }
 
-function clearInput() { //Limpar input assim que submeter
-  document.querySelector('.nome').value = "";
-  document.querySelector('.nota').value = "";
-}
\ No newline at end of file
+
+button.addEventListener("click", function(e) {
+  e.preventDefault();
```

Comment:
```suggestion
  e.preventDefault();
  handleSubmit();
```
Referente ao tópico 1, mas me perdi um pouco nos arquivos. Qualquer coisa, fala comigo depois.

Reply:
Perdão pela mistura de vários arquivos. Na verdade, eu deveria ter criado uma nova branch para cada commit que desse. Mas acredito que resolvi o que deveria ser feito.